python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -52,188 +54,211 @@ def computeClippedImageStats(im):
     return mean, std, var
 
 
-def getImageGrid(im):
-    """! Convenience function for creating a grid of pixel coordinates covering a 2-d array,
-    using `numpy.meshgrid()`
-    @param im input 2-d numpy array
-    @return two numpy arraysm each with same shape as `im`, containing x- and y- coordinate
-    grid.
-    """
-    xim = np.arange(np.int(-np.floor(im.shape[0]/2.)), np.int(np.floor(im.shape[0]/2)))
-    yim = np.arange(np.int(-np.floor(im.shape[1]/2.)), np.int(np.floor(im.shape[1]/2)))
-    x0im, y0im = np.meshgrid(xim, yim)
-    return x0im, y0im
+class ImageDecorrelationConfig(pexConfig.Config):
+    """! Configuration for image decorrelation"""
+    useScipy = pexConfig.Field(
+        dtype = bool,
+        doc = """If True use scipy for image convolution (debugging option);
+                 if False use afwMath""",
+        default = False
+    )
 
 
-def computeDecorrelationKernel(kappa, sig1=0.2, sig2=0.2):
-    """! Compute the Lupton/ZOGY post-conv. kernel for decorrelating an
-    image difference, based on the PSF-matching kernel.
-    @param kappa  A matching kernel 2-d numpy.array derived from Alard & Lupton PSF matching
-    @param sig1   Average sqrt(variance) of template image used for PSF matching
-    @param sig2   Average sqrt(variance) of science image used for PSF matching
-    @return a 2-d numpy.array containing the correction kernel
+class ImageDecorrelationTask(pipeBase.Task):
```

Comment:
Picky, but consider this name, instead of the passive form: DecorrelateImageTask